lib/mix/lib/mix/scm/git.ex

```diff
@@ -248,11 +291,7 @@ defmodule Mix.SCM.Git do
   end
 
   defp get_opts_rev(opts) do
-    if branch = opts[:branch] do
-      "origin/#{branch}"
-    else
-      opts[:ref] || opts[:tag]
-    end
+    opts[:branch] || opts[:ref] || opts[:tag]
```

Comment:
This change has the side effect to affect the output of `format/1` when `:branch` is used (old: "origin/main", new: "main").

But since `format/1` is documented to be for human consumption only, this seemed like the simplest solution to fix the `git fetch` call. It remains in line with the `format/1` callback documentation that emphasizes conciseness, and how Git itself references tags and branches indistinguishably.